### ISSUE TITLE
Mask in MaskedTensor does not change device

### ISSUE THREAD

**Issue by sandeep-189** [pytorch/pytorch#147140]
### 🐛 Describe the bug

When you create a MaskedTensor and change it to cuda, the data is the only one that change to cuda. When we use a reduction function on cuda MaskedTensor (sum, to_tensor, etc), it will always fail since the mask in on another device.

```
import torch
from torch.masked import as_masked_tensor
data = torch.tensor([1,2,3])
mask = torch.tensor([True,False,True])
mt = as_masked_tensor(data, mask).to('cuda')
mt.get_data().device, mt.get_mask().device
```
```
(device(type='cuda', index=0), device(type='cpu'))
```

### Versions

Run on torch 2.6, python 3.11

**Comment by jbschlosser** [comment]
@sandeep-189 Thanks for the report! Note that masked tensor is not currently receiving active development so it's unlikely we'll get to this soon, apologies!
